src/Sylius/Behat/Context/Ui/Admin/ManagingCountriesContext.php

```diff
@@ -82,19 +92,33 @@ public function iWantToEditThisCountry(CountryInterface $country)
     }
 
     /**
-     * @When /^I choose "([^"]*)"$/
+     * @When I choose :countryName
      */
-    public function iChoose($name)
+    public function iChoose($countryName)
     {
-        $this->countryCreatePage->chooseName($name);
+        $this->countryCreatePage->chooseNameProperly($countryName);
+    }
+
+    /**
+     * @When I add the :provinceName province with :provinceCode code
+     * @When I add the :provinceName province with :provinceCode code and :provinceAbbreviation abbreviation
+     */
+    public function iAddProvinceWithCode($provinceName, $provinceCode, $provinceAbbreviation = null)
+    {
+        $currentPage = $this->currentPageResolver->getCurrentPageWithForm($this->countryCreatePage, $this->countryUpdatePage);
+
+        $currentPage->addProvince($provinceName, $provinceCode, $provinceAbbreviation);
     }
 
     /**
      * @When I add it
+     * @When I add this country
      */
     public function iAddIt()
     {
-        $this->countryCreatePage->create();
+        $currentPage = $this->currentPageResolver->getCurrentPageWithForm($this->countryCreatePage, $this->countryUpdatePage);
+
+        $currentPage->create();
```

Comment:
`I add it` is always used on create page, on the update page we use `I same my changes`, so this change should be reverted.